src/comm/SerialLink.cc

```diff
@@ -463,12 +463,11 @@ bool SerialLink::hardwareConnect(QString &type)
     }
 
     m_port = new QSerialPort(m_portName);
-    m_port->moveToThread(this);
-
     if (!m_port) {
```

Comment:
this *m_port->moveToThread(this);* was being called before the test for a valid pointer was made.